From b2612dd9c63f8835cf2047960b8cacd464a325a4 Mon Sep 17 00:00:00 2001
From: Dirk Farin <dirk.farin@gmail.com>
Date: Wed, 12 May 2021 20:32:57 +0200
Subject: [PATCH] explicitly enable lossless mode in AOM encoder

---
 libheif/heif_encoder_aom.cc | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/libheif/heif_encoder_aom.cc b/libheif/heif_encoder_aom.cc
index 65c92b1b..65a72e29 100644
--- a/libheif/heif_encoder_aom.cc
+++ b/libheif/heif_encoder_aom.cc
@@ -47,6 +47,7 @@ struct encoder_struct_aom
   int min_q;
   int max_q;
   int threads;
+  bool lossless;
 
   aom_tune_metric tune;
 
@@ -288,6 +289,8 @@ struct heif_error aom_set_parameter_lossless(void* encoder_raw, int enable)
     encoder->max_q = 0;
   }
 
+  encoder->lossless = enable;
+
   return heif_error_ok;
 }
 
@@ -295,7 +298,7 @@ struct heif_error aom_get_parameter_lossless(void* encoder_raw, int* enable)
 {
   struct encoder_struct_aom* encoder = (struct encoder_struct_aom*) encoder_raw;
 
-  *enable = (encoder->min_q == 0 && encoder->max_q == 0);
+  *enable = encoder->lossless;
 
   return heif_error_ok;
 }
@@ -768,6 +771,9 @@ struct heif_error aom_encode_image(void* encoder_raw, const struct heif_image* i
 
   aom_codec_control(&codec, AOME_SET_TUNING, encoder->tune);
 
+  if (encoder->lossless) {
+    aom_codec_control(&codec, AV1E_SET_LOSSLESS, 1);
+  }
 
   // --- encode frame
 
