From 2dfc9b2c04ce77c0d85af37a4f66c0ee2dbe058d Mon Sep 17 00:00:00 2001
From: Dirk Farin <dirk.farin@gmail.com>
Date: Mon, 12 Jul 2021 17:00:09 +0200
Subject: [PATCH] Op_RGB_to_YCbCr: move chroma 4:2:0 sampling position to
 center (#521)

---
 libheif/heif_colorconversion.cc | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/libheif/heif_colorconversion.cc b/libheif/heif_colorconversion.cc
index 92f78b4e..f85bcccb 100644
--- a/libheif/heif_colorconversion.cc
+++ b/libheif/heif_colorconversion.cc
@@ -723,6 +723,27 @@ Op_RGB_to_YCbCr<Pixel>::convert_colorspace(const std::shared_ptr<const HeifPixel
         float g = in_g[y * in_g_stride + x];
         float b = in_b[y * in_b_stride + x];
 
+        if (subH > 1 || subV > 1) {
+          int x2 = (x + 1 < width && subH == 2 && subV == 2) ? x + 1 : x;  // subV==2 -> Do not center for 4:2:2 (see comment in Op_RGB24_32_to_YCbCr, github issue #521)
+          int y2 = (y + 1 < height && subV == 2) ? y + 1 : y;
+
+          r += in_r[y * in_r_stride + x2];
+          g += in_g[y * in_g_stride + x2];
+          b += in_b[y * in_b_stride + x2];
+
+          r += in_r[y2 * in_r_stride + x];
+          g += in_g[y2 * in_g_stride + x];
+          b += in_b[y2 * in_b_stride + x];
+
+          r += in_r[y2 * in_r_stride + x2];
+          g += in_g[y2 * in_g_stride + x2];
+          b += in_b[y2 * in_b_stride + x2];
+
+          r *= 0.25f;
+          g *= 0.25f;
+          b *= 0.25f;
+        }
+
         float cb, cr;
 
         cb = r * coeffs.c[1][0] + g * coeffs.c[1][1] + b * coeffs.c[1][2];
