From ca2473d9eca36697aa531f42209567cc663ceaee Mon Sep 17 00:00:00 2001
From: Dirk Farin <dirk.farin@gmail.com>
Date: Mon, 12 Jul 2021 20:42:05 +0200
Subject: [PATCH] check whether 'clap' input fraction exceeds supported range
 (#504)

---
 libheif/box.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libheif/box.h b/libheif/box.h
index e5019a2b..32e49290 100644
--- a/libheif/box.h
+++ b/libheif/box.h
@@ -69,6 +69,9 @@ namespace heif {
 
     Fraction(int32_t num, int32_t den);
 
+    // may only use values up to int32_t maximum
+    Fraction(uint32_t num, uint32_t den);
+
     Fraction operator+(const Fraction&) const;
 
     Fraction operator-(const Fraction&) const;
