From 9497e10168660138fd10a738179039c0e7d7ba6c Mon Sep 17 00:00:00 2001
From: Dirk Farin <dirk.farin@gmail.com>
Date: Mon, 21 Feb 2022 20:38:13 +0100
Subject: [PATCH] avoid division by zero (should never happen)

---
 libheif/nclx.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libheif/nclx.cc b/libheif/nclx.cc
index 0c35b6cd..b81b7d7c 100644
--- a/libheif/nclx.cc
+++ b/libheif/nclx.cc
@@ -82,6 +82,10 @@ heif::Kr_Kb heif::get_Kr_Kb(uint16_t matrix_coefficients_idx, uint16_t primaries
     float denom = p.whiteY * (p.redX * (p.greenY * zb - p.blueY * zg) + p.greenX * (p.blueY * zr - p.redY * zb) +
                               p.blueX * (p.redY * zg - p.greenY * zr));
 
+    if (denom == 0.0f) {
+      return result;
+    }
+
     result.Kr = (p.redY * (p.whiteX * (p.greenY * zb - p.blueY * zg) + p.whiteY * (p.blueX * zg - p.greenX * zb) +
                            zw * (p.greenX * p.blueY - p.blueX * p.greenY))) / denom;
     result.Kb = (p.blueY * (p.whiteX * (p.redY * zg - p.greenY * zr) + p.whiteY * (p.greenX * zr - p.redX * zg) +
